src/reducers/vis-state-updaters.js

```diff
@@ -1514,3 +1526,215 @@ export function updateAnimationDomain(state) {
     }
   };
 }
+
+/**
+ * Update the status of the editor
+ * @memberof visStateUpdaters
+ * @param {Object} state `visState`
+ * @param {string} mode to set to editor to
+ * @return {Object} nextState
+ */
+export const setEditorModeUpdater = (state, {mode}) => ({
+  ...state,
+  editor: {
+    ...state.editor,
+    mode
+  }
+});
+
+/**
+ * Update editor features
+ * @memberof visStateUpdaters
+ * @param {Object} state `visState`
+ * @param {[Object]} features to store
+ * @return {Object} nextState
+ */
+export function setFeaturesUpdater(state, {features = []}) {
+
+  const lastFeature = features.length && features[features.length - 1];
+
+  let newState = {
+    ...state,
+    editor: {
+      ...state.editor,
+      features,
+      mode: lastFeature && lastFeature.properties.isClosed ?
+        EDITOR_MODES.EDIT : state.editor.mode
+    }
+  };
+
+  // Retrieve existing feature
+  const {selectedFeature} = state.editor;
+
+  // If no feature is selected we can simply return since no operations
+  if (!selectedFeature) {
+    return newState;
+  }
+
+  // TODO: check if the feature has changed
+  const feature = features.find(f => f.id === selectedFeature.id);
+
+  if (!feature) {
+    return newState;
+  }
+
+  const filterIdx = getFilterIdxByFeatureId(newState, selectedFeature.id);
+
+  // no filters found
+  if (filterIdx === -1) {
+    return newState;
+  }
+
+  const filter = state.filters[filterIdx];
+
+  const newFilter = updatePolygonFilter(filter, feature);
```

Comment:
updatePolygonFilter in a nutshell updates filter.value, and all the code following it is to apply the filter.
You should be able to call  `setFilterUpdater` here with updated value, instead of all the lines below.
```
const featureInValue = {...feature, id: feature.id};
return setFilterUpdater(newState, {idx: filterIdx, prop: 'value', value: featureInValue})
```

Reply:
actually it turns out if we call setFilterUpdater we don't need to use updatePolygonFilter and we can simply do the following:
```
return setFilterUpdater(newState, {idx: filterIdx, prop: 'value', value: feature});
```

Reply:
no longer valid